### PR TITLE
refactor towards proposed interface change.

### PR DESCRIPTION
**Pull request by xiaoyunwu**

### CODE REVIEW

File: framework/data_request.go [hunk 1]
```diff
@@ -107,9 +107,11 @@ func (f *framework) handleDataReq(dr *dataRequest) {
 	dataReceiver := make(chan []byte, 1)
 	switch {
 	case topoutil.IsParent(f.topology, dr.epoch, dr.taskID):
-		f.task.ServeAsChild(dr.taskID, dr.req, dataReceiver)
+		b, _ := f.task.ServeAsChild(dr.taskID, dr.req)
```
Comment (fengjingchao) [inline framework/data_request.go]: Panic if error is returned.

File: framework_interface.go [hunk 2]
```diff
@@ -40,17 +40,20 @@ type Framework interface {
 // Context is used in task callbacks. It provides APIs for tasks to ask framework
 // to do work in certain context.
 type Context interface {
-	// These two are useful for task to inform the framework their status change.
+	// This useful for task to inform the framework their status change.
```
Comment (fengjingchao) [inline framework_interface.go]: This is

File: framework/data_request.go (hunk 1 above)
Comment (fengjingchao) [inline framework/data_request.go]:
```
if err != nil {
  f.logger.Panicf("Task ServeAsChild failed: %v", err)
}
```

**Comment by fengjingchao** [comment]
@xiaoyunwu 
Can you explain why Task implementation ServeAsX should return error? What should framework do if it happened.

**Comment by xiaoyunwu** [comment]
func (s *routeGuideServer) GetFeature(ctx context.Context, point *pb.Point)
(*pb.Feature, error) { for _, feature := range s.savedFeatures { if proto.
Equal(feature.Location, point) { return feature, nil } } // No feature was
found, return an unnamed feature return &pb.Feature{"", point}, nil} when
we move to rpc, it will generate error. since it is rpc, the error will
pass to client directly. so we do have to worry.

**Comment by xiaoyunwu** [comment]
do not have to worry.

**Comment by xiaoyunwu** [comment]
see also
https://github.com/grpc/grpc-go/blob/master/examples/route_guide/client/client.go#L63

**Comment by fengjingchao** [comment]
LGTM

**Pull request merged**